### PR TITLE
Introduction of credentials as API authentication method

### LINKED ISSUE #147: Add srun to scratch_copy in config

### PR DESCRIPTION
**Pull request by Shettland**
I introduced user and password credentials to authenticate the requests to iskylims API into all the modules.

The main differences are present in:
1. [__main__.py](https://github.com/BU-ISCIII/buisciii-tools/blob/develop/bu_isciii/__main__.py). Buisciii-tools can now be executed using a credentials file in yaml format in the user's home directory; or specify them before the command that is going to be passed to the Command line.
2. [utils.py](https://github.com/BU-ISCIII/buisciii-tools/blob/develop/bu_isciii/utils.py) Received three functions that aim to extract the fields from the yaml file and to verify that the extraction was correct.

The rest of the modules received minor changes such as error handling.

I also included the srun command into the scratch module so the copy process is done using SLURM's queue manager.

Closes #147

### CODE REVIEW

**Comment by Shettland** [comment]
Black-linter is failing when running it with github actions. Found an issue reporting the same problem:
https://github.com/psf/black/issues/3953

File: bu_isciii/utils.py [hunk 1]
```diff
@@ -387,5 +391,54 @@ def ask_date(previous_date=None, posterior_date=None, initial_year=2010):
     return datetime.date(int(year), int(chosen_month_number), int(day))
 
 
+def get_yaml_config():
+    """Extract the fields from the config yaml file described in configuration.json"""
```
Comment (svarona) [inline bu_isciii/utils.py:395]: Use VSC Generate Docstring

File: bu_isciii/scratch.py [hunk 2]
```diff
@@ -99,20 +99,27 @@ def __init__(
...
+        stderr.print("[blue]to %s" % self.scratch_tmp_path)
         if self.service_folder in self.full_path:
             protocol = self.conf["protocol"]
             try:
                 if protocol == "rsync":
-                    sysrsync.run(
+                    rsync_command = sysrsync.get_rsync_command(
                         source=self.full_path,
-                        destination=self.tmp_dir,
+                        destination="/scratch/bi/",
```
Comment (svarona) [inline bu_isciii/scratch.py:117]: La ruta "/scratch/bi/" que la coja siempre del config.json por si cambia, creala como variable en scratch_copy, como scratch_path o algo así, igual que se hace con /data/bi y /srv/fastq_repo

Comment (Shettland) [reply bu_isciii/scratch.py:117]: Ok

**Review by svarona — APPROVED** [review]

**Pull request merged**